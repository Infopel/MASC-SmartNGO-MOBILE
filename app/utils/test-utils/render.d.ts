export type RenderOptions = {
  wrapper?: React.ComponentType<any>
  createNodeMock?: (element: React.ReactElement) => any
}
export type AllTheProvidersProps = { children: React.ReactNode }
