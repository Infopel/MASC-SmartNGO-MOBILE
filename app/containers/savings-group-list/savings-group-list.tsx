import { withDatabase } from "@nozbe/watermelondb/DatabaseProvider"
import withObservables from "@nozbe/with-observables"
import { BaseList } from "components/base-list/base-list"
import { SavingsGroupItem } from "components/savings-group-item/savings-group-item"
import { observer } from "mobx-react-lite"
import { FlatList, IBoxProps } from "native-base"
import { navigate } from "navigators/navigation-utilities"
import * as React from "react"
import { findAllSavingsGroups, ISavingsGroup } from "storage/queries/savings-group"

export interface SavingsGroupListProps extends IBoxProps {
  items: ISavingsGroup[]
}

/**
 * Describe your component here
 */
export const Container = observer(function SavingsGroupList(props: SavingsGroupListProps) {
  const { items, ...rest } = props
  const handleOnPress = (id: string) => {
    navigate('savings-group-details', { id})
  }
  return (
    <BaseList data={items} renderItem={({ item }) => <SavingsGroupItem onPress={handleOnPress.bind(this,item.id)} item={item} />} />
  )
})


export const SavingsGroupListContainer = withDatabase(withObservables(['database'], ({ database }) => ({ items: findAllSavingsGroups(database) }))(Container))