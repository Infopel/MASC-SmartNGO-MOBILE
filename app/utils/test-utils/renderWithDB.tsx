jest.mock("storage/adapter.ts")
import DatabaseProvider from "@nozbe/watermelondb/DatabaseProvider"
import { render } from "@testing-library/react-native"
import { default as React } from "react"
import { database } from "storage/database"
import { NativeBaseProvider as ThemeProvider } from "theme"
import { AllTheProvidersProps, RenderOptions } from "./render"

function AllTheProvidersWithDB({ children }: AllTheProvidersProps) {
    return (
        <DatabaseProvider database={database}>
            <ThemeProvider>{children}</ThemeProvider>
        </DatabaseProvider>
    )
}

export function renderWithDB<T extends any>(ui: React.ReactElement<T>, options?: RenderOptions) {
    return render(ui, { wrapper: AllTheProvidersWithDB, ...options })
}


export class DiagnosticError extends Error {
    name: string = "Diagnostic error"
  }
  