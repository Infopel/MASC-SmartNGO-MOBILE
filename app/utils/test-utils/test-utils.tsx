import { render } from "@testing-library/react-native"
import React from "react"
import { NativeBaseProvider as ThemeProvider } from "theme"
import { AllTheProvidersProps, RenderOptions } from "./render"

function AllTheProviders({ children }: AllTheProvidersProps) {
  return (
    <ThemeProvider>{children}</ThemeProvider>
  )
}

function customRender<T extends any>(ui: React.ReactElement<T>, options?: RenderOptions) {
  return render(ui, { wrapper: AllTheProviders, ...options })
}

// re-export everything
export * from "@testing-library/react-native"
// override render method
export { customRender as render }

