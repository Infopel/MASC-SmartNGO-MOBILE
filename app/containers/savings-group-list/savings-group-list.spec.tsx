import { fireEvent, render } from "utils/test-utils/test-utils"
import * as React from "react"
import { SavingsGroupListContainer } from "./savings-group-list"

describe("<SavingsGroupListContainer>",()=>{
 it("render", async function () {
    const component = <SavingsGroupListContainer></SavingsGroupListContainer>
    
    expect(render(component).toJSON()).toMatchSnapshot()
  })
})
  