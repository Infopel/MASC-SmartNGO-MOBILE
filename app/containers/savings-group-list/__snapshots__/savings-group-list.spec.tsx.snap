// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SavingsGroupListContainer> render 1`] = `
<RCTScrollView
  contentContainerStyle={
    Array [
      Object {},
      Object {
        "dataSet": Object {},
      },
    ]
  }
  data={
    Array [
      Object {
        "cycleDuration": 35,
        "id": "s",
        "location": "Maputo Cidade, 25 de Setembro",
        "name": "Poupança",
        "numberOfCycles": 8,
        "numberOfMembers": 15,
      },
    ]
  }
  dataSet={Object {}}
  getItem={[Function]}
  getItemCount={[Function]}
  keyExtractor={[Function]}
  onContentSizeChange={[Function]}
  onLayout={[Function]}
  onMomentumScrollBegin={[Function]}
  onMomentumScrollEnd={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  removeClippedSubviews={false}
  renderItem={[Function]}
  scrollEventThrottle={50}
  stickyHeaderIndices={Array []}
  style={Object {}}
  viewabilityConfigCallbackPairs={Array []}
>
  <View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <View
        dataSet={Object {}}
        style={
          Object {
            "backgroundColor": "#fafaf9",
            "borderBottomColor": "#fafafa",
            "borderLeftColor": "#fafafa",
            "borderRadius": 6,
            "borderRightColor": "#fafafa",
            "borderTopColor": "#fafafa",
            "borderWidth": 1,
            "elevation": 2,
            "shadowColor": "#000000",
            "shadowOffset": Object {
              "height": 1,
              "width": 0,
            },
            "shadowOpacity": 0.2,
            "shadowRadius": 1.41,
          }
        }
      >
        <View
          _icon={
            Object {
              "color": "primary.600",
              "size": "sm",
            }
          }
          _spinner={
            Object {
              "color": "primary.600",
              "focusable": false,
              "size": "sm",
            }
          }
          _stack={
            Object {
              "alignItems": "center",
              "space": "1.5",
            }
          }
          _text={
            Object {
              "color": "primary.600",
              "fontSize": "sm",
            }
          }
          accessible={true}
          collapsable={false}
          colorScheme="primary"
          dataSet={Object {}}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "borderRadius": 4,
              "flexDirection": "row",
              "justifyContent": "center",
              "paddingBottom": 10,
              "paddingLeft": 12,
              "paddingRight": 12,
              "paddingTop": 10,
            }
          }
        >
          <View
            dataSet={Object {}}
            style={
              Object {
                "flexDirection": "row",
              }
            }
          >
            <View
              dataSet={Object {}}
              style={
                Object {
                  "flex": 1,
                }
              }
            >
              <Text
                dataSet={Object {}}
                style={
                  Object {
                    "backgroundColor": undefined,
                    "color": "#171717",
                    "fontFamily": undefined,
                    "fontSize": 14,
                    "fontStyle": "normal",
                    "fontWeight": "700",
                    "letterSpacing": 0,
                    "lineHeight": 21,
                    "textDecorationLine": undefined,
                  }
                }
              >
                Poupança
              </Text>
              <View
                dataSet={Object {}}
                style={
                  Object {
                    "height": 8,
                  }
                }
              />
              <Text
                dataSet={Object {}}
                style={
                  Object {
                    "backgroundColor": undefined,
                    "color": "#171717",
                    "fontFamily": undefined,
                    "fontSize": 14,
                    "fontStyle": "normal",
                    "fontWeight": "400",
                    "letterSpacing": 0,
                    "lineHeight": 21,
                    "textDecorationLine": undefined,
                  }
                }
              >
                Maputo Cidade, 25 de Setembro
              </Text>
              <View
                dataSet={Object {}}
                style={
                  Object {
                    "height": 8,
                  }
                }
              />
              <View
                dataSet={Object {}}
                style={
                  Object {
                    "flexDirection": "row",
                  }
                }
              >
                <Text
                  dataSet={Object {}}
                  style={
                    Object {
                      "backgroundColor": undefined,
                      "color": "#171717",
                      "fontFamily": undefined,
                      "fontSize": 14,
                      "fontStyle": "normal",
                      "fontWeight": "400",
                      "letterSpacing": 0,
                      "lineHeight": 21,
                      "textDecorationLine": undefined,
                    }
                  }
                >
                  Cycle duration: 
                </Text>
                <View
                  dataSet={Object {}}
                  style={Object {}}
                />
                <Text
                  dataSet={Object {}}
                  style={
                    Object {
                      "backgroundColor": undefined,
                      "color": "#171717",
                      "fontFamily": undefined,
                      "fontSize": 14,
                      "fontStyle": "normal",
                      "fontWeight": "400",
                      "letterSpacing": 0,
                      "lineHeight": 21,
                      "textDecorationLine": undefined,
                    }
                  }
                >
                  35
                </Text>
              </View>
              <View
                dataSet={Object {}}
                style={
                  Object {
                    "height": 8,
                  }
                }
              />
              <View
                dataSet={Object {}}
                style={
                  Object {
                    "flexDirection": "row",
                  }
                }
              >
                <Text
                  dataSet={Object {}}
                  style={
                    Object {
                      "backgroundColor": undefined,
                      "color": "#171717",
                      "fontFamily": undefined,
                      "fontSize": 14,
                      "fontStyle": "normal",
                      "fontWeight": "400",
                      "letterSpacing": 0,
                      "lineHeight": 21,
                      "textDecorationLine": undefined,
                    }
                  }
                >
                  8
                </Text>
                <View
                  dataSet={Object {}}
                  style={Object {}}
                />
                <Text
                  dataSet={Object {}}
                  style={
                    Object {
                      "backgroundColor": undefined,
                      "color": "#171717",
                      "fontFamily": undefined,
                      "fontSize": 14,
                      "fontStyle": "normal",
                      "fontWeight": "400",
                      "letterSpacing": 0,
                      "lineHeight": 21,
                      "textDecorationLine": undefined,
                    }
                  }
                >
                   Cycles
                </Text>
              </View>
            </View>
            <View
              dataSet={Object {}}
              style={Object {}}
            />
            <View
              dataSet={Object {}}
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#a1a1aa",
                  "borderColor": "#27272a",
                  "borderRadius": 9999,
                  "color": "#f97316",
                  "height": 48,
                  "justifyContent": "center",
                  "position": "relative",
                  "width": 48,
                }
              }
            >
              <Text
                dataSet={Object {}}
                style={
                  Object {
                    "backgroundColor": undefined,
                    "color": "#fafafa",
                    "fontFamily": undefined,
                    "fontSize": 16,
                    "fontStyle": "normal",
                    "fontWeight": "600",
                    "letterSpacing": 0,
                    "lineHeight": 24,
                    "textDecorationLine": undefined,
                  }
                }
              >
                8
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
